internal/datastore/crdb/tx.go

```diff
@@ -61,10 +64,17 @@ func executeWithResets(ctx context.Context, fn innerFunc, maxRetries uint8) (err
 	for retries = 0; retries <= maxRetries; retries++ {
 		err = fn(ctx)
 		if resettable(ctx, err) {
-			log.Ctx(ctx).Warn().Err(err).Msg("retrying resetteable database error")
+			after := retry.BackoffExponentialWithJitter(100*time.Millisecond, 0.5)(uint(retries))
+			log.Ctx(ctx).Warn().Err(err).Dur("after", after).Msg("retrying resetteable database error")
+			time.Sleep(after)
 			continue
 		}
-
+		if err != nil {
+			log.Ctx(ctx).Warn().Err(err).Msg("non-resettable database error")
```

Comment:
do we need to log this? won't the error propagate up the stack and eventually get logged? maybe we can wrap the error with additional context if we want to add that information